\chapter{Introduction}\label{intro}
\index{Introduction@\emph{Introduction}}%
\epigraph{Nothing in biology makes sense except in the light of evolution}{Christian Theodosius Dobzhansky}

Evolution is the cornerstone of modern biology.  Through evolution, we have answered questions in areas such as hominid and human origins~\cite{Martin1990,Takahata1997}, vaccination development~\cite{Wilder-Smith2010,Fitch1993}, and even in environmental cleanup using microbes~\cite{Liu1993}.  \emph{Phylogenetics}, the study of the evolutionary relationships between different organisms, is vital in answering many of these questions.  

One of the most ambitious projects in phylogenetics is the Assembling the Tree of Life (AToL) project~\cite{atol-website}.  The goal of the AToL project is a tree representing the evolutionary relationship between all species on Earth.  As there are an estimated 9 million species on Earth~\cite{Mora2011}, one can imagine that determining the relationship between such a huge number of diverse species can be extremely challenging, and that efficient and accurate methods are needed for this task.

A typical pipeline for inferring the relations between different species is to collect biomolecular sequences from the species (DNA, RNA, or amino acid sequences), and from those sequences, estimate a \emph{Multiple Sequence Alignment} (MSA) containing the sequences.  The MSA is a hypothesis of the evolutionary history between the different characters in the sequences.  From the MSA, a tree representing the relationship between the species can be inferred.  This pipeline can be computationally expensive if each step (MSA estimation and tree building) has to be redone every time a new species is sequenced.

Imagine that for a particular set of species, an MSA and a tree has already been constructed from the sequences of those species, and a new sequence has now been collected from a novel species.  Instead of re-building the MSA and tree from scratch, an alternative approach is to insert the new sequence into the existing MSA.  This produces an alignment of the original sequences plus the new sequence (called ``extended alignment'').  The extended alignment can then be used to place the sequence into the existing tree.  The placement location can then used to infer the relationship between the new species and the original species.  

The problem I described is known as the \emph{phylogenetic placement problem}.  Given an existing MSA (called the ``backbone alignment''), an existing tree (called the ``backbone tree''), and a new sequence (called the ``query sequence''), how does one insert the query sequence into the backbone tree.  My investigation into this problem has lead to the development of a new machine learning technique which I call the families of Hidden Markov Models (fHMMs).  The fHMMs is a technique that allows accurate alignment of the query sequence to an existing backbone alignment.  This technique can be applied to a wide range of problems that uses alignments, including determining the evolutionary relationship between different species, taxonomically classifying short DNA fragments from environmental samples, estimating alignments on extremely large datasets.

In Chapter~\ref{background}, I formally introduce key concepts in phylogenetics such as MSA estimation and tree estimation.  I also introduce the use of HMMs for aligning query sequences to an MSA.  Finally, I introduce 3 problems that will be addressed using the fHMMs technique: phylogenetic placement, taxonomic profiling and taxonomic identification, and MSA estimation.  In Chapter~\ref{hmmfamily}, I describe the fHMM technique.  In Chapter~\ref{sepp_chapter}, I present a simulation study on using fHMMs to address the phylogenetic placement problem.  I show that the fHMMs technique results in more accurate placements than using a single HMM, and can accurately place sequences that are very evolutionarily divergent.  In Chapter~\ref{tipp_chapter}, I show a modification of the fHMM technique for the problems of taxonomic identification and profiling.  By incorporating statistical support in the fHMM technique, the precision in taxonomically classifying novel sequences is greatly enhanced.  In addition, I show that fHMM technique results in better estimations of the species abundance profile of simulated microbial communities.  In Chapter ~\ref{upp_chapter}, I show a simple modification of the fHMMs that allows for ``de novo'' MSA estimation of ultra-large datasets without the need of an initial backbone alignment and tree.  I show that this method is a fast and efficient MSA estimation method, and results in more accurate MSAs compared to other MSA methods.  I show that this new technique can accurate align a dataset of 1,000,000 sequences in less than 2 days without the need of a supercomputer.  Finally, in Chapter~\ref{conclusion}, I summarize the contributions of this dissertation and discuss future work.
