\chapter{Introduction}
\index{Introduction@\emph{Introduction}}%
\emph{Phylogenetics} is the study of the evolutionary relationship between different organisms.  The output of a phylogenetic study is a \emph{phylogeny}, or the evolutionary history of the different organisms through time.  A simple representation of a phylogeny is a tree.  Each internal node in the tree represents a speciation event, one species giving rise to new lineages of species.  The leaves of the phylogeny represents current species, both extant and extinct.  Typically, the goal in a phylogenetic study is to reconstruct the phylogeny given the leaves.

The pipeline in a typical phylogenetic study begins by first collecting the biomolecular sequences (DNA, RNA, or protein) of the species in question.  Next, a \emph{multiple sequence alignment} is estimated from the sequences.  The alignment is a hypothesis about the \emph{homology}, or shared evolutionary history, between the sequences.  From the alignment, a phylogeny can be estimated.  

The increasing size of the data, in terms of the number of sequences, presents new challenges in the pipeline.  Previous Sanger shotgun sequencing technology produced very long reads that were assembled into a full length sequences by examining the regions of overlap.  Next generation sequencing (NGS) technologies uses different processes to create millions of short reads very rapidly.  This process is performed in parallel, allowing very fast sequencing of entire genomes.  Thus, the number of sequences in a phylogenetic study can be very large as more and more genomes are being sequenced.

One downside of NGS is the shortness of the reads.  Since NGS reads are shorter than Sanger reads, assembly is more difficult, and sometimes, may not be possible due to insufficient coverage of the genome, insufficient overlap of the reads, or the source of the reads is unknown.  This leads to mixed datasets, datasets containing both full length sequences and short reads.  Mixed dataset are difficult to analyze using traditional approaches.  Mixed datasets may violate the assumptions of traditional alignment and phylogeny estimation tools.  In addition, mixed datasets may contain millions of reads making traditional tools computationally infeasible. Modifications of existing algorithms are necessary to be able to analyze mixed datasets efficiently and accurately.

An example of when mixed datasets arise is in \emph{metagenomic} studies.  In a metagenomic study, the genetic material are taken directly from the environment and sequenced.  The short reads come from the various known and unknown species in the sample.  Without knowing the source species from which the read originated from, assembly is very difficult.  One of the fundamental steps in a metagenomic study is to identify the taxa of the reads.  This is known as the \emph{taxonomic identification problem}.  %The goal is to find the taxonomic classification of each short read.

Phylogenetic placement is one method for analyzing mixed datasets.  The phylogenetic placement takes as input a set of query sequences (short reads or full length sequences) and a backbone alignment and tree.  The output is the optimal placement of each query sequence into the backbone tree.  Phylogenetic placement typically has two steps.  First, align the query sequence to the backbone alignment.  Second, use the extended alignment to place the query sequence into the backbone tree.  The placement of the query sequence is a hypothesis of the evolutionary relationship between the query sequence and the remaining sequences in the backbone tree.  The placement of the query sequence can be used to for taxonomic identification, among other uses.

In Chapter~\ref{background}, I formally introduce alignment and phylogeny estimation and describe current methods for these steps.  I also introduce the use of Hidden Markov Models (HMMs) for multiple sequence alignment.  Finally, I describe concepts in metagenomic analyses, namely taxonomic identification and profiling.  In Chapter ~\ref{sepp_chapter}, I present SEPP, a new method for phylogenetic placement using families of HMMs.  I show that SEPP improves upon the use of a single HMM, and present results showing that SEPP has better accuracy than two popular techniques.  In Chapter~\ref{tipp_chapter}, I show that SEPP results in a high false positive rate for taxonomic identification.  I then present TIPP, a modification of SEPP for taxonomic identification and profiling.  TIPP incorporates statistical support measures to greatly reduce the false classification rate of SEPP.  I show results that TIPP classifies more fragments correctly compared to other taxonomic identification methods, and that TIPP results in overall better profiles.  In Chapter ~\ref{upp_chapter}, I show a simple modification of SEPP called UPP that allows for the alignment of ultra-large datasets.  I show that UPP is a fast and efficient alignment method and results in more accurate alignments compared to other large alignment methods.  I show that UPP can accurate align a dataset of 1,000,000 sequences in less than 2 days.  Finally, in Chapter~\ref{conclusion}, I summarize the contributions of this dissertation and discuss future work.